### ISSUE TITLE
Manual validation run for room owners

### ISSUE THREAD

**Issue by RePod** [Eijebong/Archipelago-lobby#33]
When rooms that have validation disabled allow a manual validation run to catch surface level problems.
For example, the [latest Hades release](https://github.com/NaixGames/Polycosmos/releases/tag/0.12.3) provides *template YAMLs* with outdated key names. Yes, this would be caught if the submitter solo genned *and* if the room had validation enabled, but this mitigates a downside of disabling validation(...) and can inform room owners ahead of time.

Easy mode:
 - Ignore/warn games not tracked by the lobby
 - Rate limit of course

Easy NG+ mode:
 - Only check YAMLs that have appeared/changed since last manual run
   - #29 
   - Maybe near the button to start it: "X changes since last run"

**Comment by Eijebong** [comment]
It is something that's been on my mind for a while because of lobby generation. Here's my braindump about it from a few weeks ago:


1. I add a validation status to YAMLs.

    - Validated (it's been validated by the checker itself, I know for sure it's good)
    - Manually validated (it's been vetted by the author of the room, it might be bad but at least someone should've looked at it...)
    - Non validated
3. I add a colored circle next to YAMLs for the room author. Green if validated, light green if manually validated, red if non validated. IIF they have validation on, otherwise don't care.
4. Depending on the YAML validation state:
    - If it's Validated clicking that circle does nothing
    - If it's Manually validated, give an option to invalidate it and an option to throw it at the YAML checker.
    - If it's Non validated, give an option to throw it at the checker and an option to mark it as Manually validated
5. In the room options, I change a few settings:
    - I remove/change the option to enable validation or not:
    - I add an APWorlds that shouldn't get validated by the lobby, a free form list that autocompletes with apworlds present in the room. When a YAML matches that, I just set their status as Non validated and upload them.
    - I add an option to not validate anything
    - I add an option only emit a warning when validation fails
    - I add a Forbidden apworlds, same kind of list. Those would not be allowed in the room at all.

**Comment by Eijebong** [comment]
Yup, that was a brain dump I made, haven't re-read any of it but your input is very welcome, thank you

**Comment by Eijebong** [comment]
A few more thoughts about this:

We can automatically revalidate YAMLs when an apworld changes in the room in 2 places.
1. When the room gets saved and is open
2. After the lobby refreshes its index, on all open rooms

The main issue with that is knowing what to revalidate. The validation status should probably contain the apworlds/versions the YAML has been checked with.
